rust/agama-server/src/network/nm/client.rs

```diff
@@ -288,16 +302,24 @@ impl<'a> NetworkManagerClient<'a> {
     /// * `path`: D-Bus patch of the connection.
     async fn deactivate_connection(&self, path: OwnedObjectPath) -> Result<(), ServiceError> {
         let proxy = NetworkManagerProxy::new(&self.connection).await?;
-        match proxy.deactivate_connection(&path.as_ref()).await {
-            Err(e) => {
-                // Ignore ConnectionNotActive error since this just means the state is already correct
-                if e.to_string().contains("ConnectionNotActive") {
-                    Ok(())
-                } else {
-                    Err(ServiceError::DBus(e))
+
+        if let Some(active_connection) = self.settings_active_connection(path.clone()).await? {
```

Comment:
What about something like:

```rust
        if let Some(active_connection) = self.settings_active_connection(path.clone()).await? {
            if let Err(e) = proxy
                .deactivate_connection(&active_connection.as_ref())
                .await
            {
                // Ignore ConnectionNotActive error since this just means the state is already correct
                if !e.to_string().contains("ConnectionNotActive") {
                    return Err(ServiceError::DBus(e));
                }
            }
        }
        Ok(())
```